pkg/controller/deployment/util/deployment_util.go

```diff
@@ -508,7 +507,7 @@ func getReplicaSetFraction(logger klog.Logger, rs apps.ReplicaSet, d apps.Deploy
 	// We should never proportionally scale up from zero which means rs.spec.replicas and annotatedReplicas
 	// will never be zero here.
 	newRSsize := (float64(*(rs.Spec.Replicas) * deploymentReplicas)) / float64(annotatedReplicas)
-	return integer.RoundToInt32(newRSsize) - *(rs.Spec.Replicas)
+	return int32(math.Round(newRSsize)) - *(rs.Spec.Replicas)
```

Comment:
This change does not belong here. Please don't do this only for the sake of removing `k8s.io/utils/integer` package.

Reply:
Thanks.
Got it and fixed.